File: java/ceasar-cipher-encryption.java


import java.util.Scanner;

public class CaesarEncryption {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter plain text: ");
        String text = sc.nextLine();
        System.out.print("Enter shift value (0-25): ");
        int shift = sc.nextInt();
        sc.close();

        String encrypted = encrypt(text, shift);
        System.out.println("\n🔒 Encrypted Text: " + encrypted);
    }

    public static String encrypt(String text, int shift) {
        StringBuilder result = new StringBuilder();
        for (char ch : text.toCharArray()) {
            if (Character.isLetter(ch)) {
                char base = Character.isUpperCase(ch) ? 'A' : 'a';
                ch = (char) ((ch - base + shift) % 26 + base);
            }
            result.append(ch);
        }
        return result.toString();
    }
}
// Enter plain text: Hello World
// Enter shift value (0-25): 3
// 🔒 Encrypted Text: Khoor Zruog

File: python /ceasar-cipher-encryption.py


def caesar_encrypt(plain_text, shift):
    encrypted_text = ""
    for char in plain_text:
        if char.isalpha():
            # Determine ASCII base (A=65 for uppercase, a=97 for lowercase)
            base = 65 if char.isupper() else 97
            # Shift forward by 'shift' and wrap around using modulo 26
            encrypted_char = chr((ord(char) - base + shift) % 26 + base)
            encrypted_text += encrypted_char
        else:
            # Leave non-letters unchanged
            encrypted_text += char
    return encrypted_text


# --- Example Usage ---
text = input("Enter plain text: ")
shift = int(input("Enter shift value (0-25): "))

encrypted = caesar_encrypt(text, shift)
print("\n🔒 Encrypted Text:", encrypted)


# Input:
# Enter plain text: Hello World
# Enter shift value (0-25): 3

# Output:
# 🔒 Encrypted Text: Khoor Zruog